Model/SourcesRepository.php

```
@@ -0,0 +1,178 @@
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var \Magento\Framework\Api\DataObjectHelper
+     */
+    protected $dataObjectHelper;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultFactory;
+
+    /**
+     * @var SourceSelectionItemInterfaceFactory
+     */
+    private $sourceSelectionItemInterface;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SerializerInterface $serializer
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory
+     * @param SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+     */
+    public function __construct(
+        SourcesFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SerializerInterface $serializer,
+        \Magento\Framework\Api\DataObjectHelper $dataObjectHelper,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory,
+        SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->serializer = $serializer;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourceSelectionItemInterface = $sourceSelectionItemInterface;
+
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws NoSuchEntityException
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
```

Yo, can we throw a not found error here if the source extension entry does not exist?